src/select-menu/src/OptionsList.js

```diff
@@ -139,7 +139,7 @@ const OptionsList = memo(function OptionsList(props) {
 
   const handleSelect = useCallback(
     item => {
-      if (isSelected(item)) {
+      if (isSelected(item) && isMultiSelect) {
```

Comment:
Only de-select an item if we're in multiselect, otherwise this is a no-op, and we should just "select" regardless -- we have an effect hook that checks if select ever goes out of date, and updates the `OptionsList`s internal state.